File: backend/app/routers/patient.py
from fastapi import APIRouter, Depends, HTTPException
from sqlalchemy.orm import Session
from typing import List
from app.database import get_db
from app.models.patient import Patient
from app.schemas.schemas import PatientCreate, PatientUpdate, PatientOut

router = APIRouter(prefix="/api/patients", tags=["patients"])


@router.get("/", response_model=List[PatientOut])
def list_patients(db: Session = Depends(get_db)):
    return db.query(Patient).order_by(Patient.admission_date.desc()).all()


@router.post("/", response_model=PatientOut)
def create_patient(data: PatientCreate, db: Session = Depends(get_db)):
    patient = Patient(**data.model_dump())
    db.add(patient)
    db.commit()
    db.refresh(patient)
    return patient


@router.get("/{id}", response_model=PatientOut)
def get_patient(id: int, db: Session = Depends(get_db)):
    patient = db.query(Patient).filter(Patient.id == id).first()
    if not patient:
        raise HTTPException(404, "患者不存在")
    return patient


@router.put("/{id}", response_model=PatientOut)
def update_patient(id: int, data: PatientUpdate, db: Session = Depends(get_db)):
    patient = db.query(Patient).filter(Patient.id == id).first()
    if not patient:
        raise HTTPException(404, "患者不存在")
    for k, v in data.model_dump(exclude_unset=True).items():
        setattr(patient, k, v)
    db.commit()
    db.refresh(patient)
    return patient


@router.delete("/{id}")
def delete_patient(id: int, db: Session = Depends(get_db)):
    patient = db.query(Patient).filter(Patient.id == id).first()
    if not patient:
        raise HTTPException(404, "患者不存在")
    db.delete(patient)
    db.commit()
    return {"message": "已删除"}

File: frontend/src/stores/patient.js
import { defineStore } from 'pinia'
import api from '../api'

export const usePatientStore = defineStore('patient', {
  state: () => ({
    patients: [],
    currentPatient: null,
  }),
  actions: {
    async fetchPatients() {
      const res = await api.getPatients()
      this.patients = res.data
    },
    async fetchPatient(id) {
      const res = await api.getPatient(id)
      this.currentPatient = res.data
    },
    async createPatient(data) {
      const res = await api.createPatient(data)
      this.patients.push(res.data)
      return res.data
    },
    async updatePatient(id, data) {
      const res = await api.updatePatient(id, data)
      const idx = this.patients.findIndex(p => p.id === id)
      if (idx >= 0) this.patients[idx] = res.data
      return res.data
    },
    async deletePatient(id) {
      await api.deletePatient(id)
      this.patients = this.patients.filter(p => p.id !== id)
    },
  },
})